web/src/javascript/drawMap.js

```diff
@@ -22,27 +21,48 @@ const miniMapPosX = screenWidth-miniMapWidth;
 const miniMapScreenPosWidth = miniMapWidth/3;
 const miniMapScreenPosHeight = miniMapHeight/3;
 
-export function drawCellsOnMap(cells){
-  //set dimensions
-  mapCanvas.width = mapWidth;
-  mapCanvas.height = mapHeight;
+export function createGameCanvas() {
+  return document.createElement("canvas");
+}
 
-  for(let i = 0; i < cells.length; i += 1) {
-    const cell = cells[i];
+export function initMap(canvas) {
+  //set dimensions
+  canvas.width = mapWidth;
+  canvas.height = mapHeight;
+}
 
-    mapContext.beginPath();
-    mapContext.arc(cell.position.x, cell.position.y, cell.mass, 0, Math.PI * 2, false);
-    mapContext.fillStyle = "#ed1515";
-    mapContext.fill();
+export function drawCellsOnMap(cells, canvas) {
+  const context = canvas.getContext("2d");
+  for(const cell of cells) {
+    context.beginPath();
+    context.arc(cell.position.x, cell.position.y, cell.mass, 0, Math.PI * 2, false);
+    context.fillStyle = "#ed1515";
+    context.fill();
   }
 }
 
+export function drawFoodOnMap(foods, canvas) {
+  const context = canvas.getContext("2d");
+  const drawFood = (foods, color) => {
+    for(const food of foods) {
+      context.beginPath();
+      context.arc(food.x, food.y, constants.foodMass, 0, Math.PI * 2, false);
```

Comment:
An idea: a `drawCircle` function with all the `beginPath`/`arc` stuff? (cells, food, viruses, just so that we don't repeat too much).